src/components/queryBuilder/views/TableQueryBuilder.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from 'react';
+import { ColumnsEditor } from '../ColumnsEditor';
+import { AggregateColumn, BuilderMode, Filter, TableColumn, OrderBy, QueryBuilderOptions, SelectedColumn, AggregateType } from 'types/queryBuilder';
+import { OrderByEditor } from '../OrderByEditor';
+import { LimitEditor } from '../LimitEditor';
+import { FiltersEditor } from '../FilterEditor';
+import allLabels from 'labels';
+import { ModeSwitch } from '../ModeSwitch';
+import { AggregateEditor } from '../AggregateEditor';
+import { GroupByEditor } from '../GroupByEditor';
+
+interface TableQueryBuilderProps {
+  allColumns: TableColumn[];
+  builderOptions: QueryBuilderOptions,
+  onBuilderOptionsChange: (builderOptions: QueryBuilderOptions) => void;
+}
+
+const emptyAggregate: AggregateColumn = { column: '', aggregateType: AggregateType.Count };
+
+export const TableQueryBuilder = (props: TableQueryBuilderProps) => {
+  const { allColumns, builderOptions, onBuilderOptionsChange } = props;
+  const [isAggregateMode, setAggregateMode] = useState<boolean>(false);
+  const [selectedColumns, setSelectedColumns] = useState<SelectedColumn[]>([]);
+  const [aggregates, setAggregates] = useState<AggregateColumn[]>([emptyAggregate]);
+  const [groupBy, setGroupBy] = useState<string[]>([]);
+  const [orderBy, setOrderBy] = useState<OrderBy[]>([]);
+  const [limit, setLimit] = useState<number>(100);
+  const [filters, setFilters] = useState<Filter[]>([]);
+  const labels = allLabels.components.TableQueryBuilder;
+
+  useEffect(() => {
```

Comment:
Yes, this hook is common with the `views/` components and is for loading the component's local state, usually from a saved query. Whenever the options are changed it updates the main builder options. It's a bit easier to reason with the builder options when they are in the context of a specific editor. For the more complicated editors (such as traces/logs) it makes it easier to edit the data before pushing it to the top level query state. I think this could still be accomplished without using the local `useState` calls, but it would be harder to follow what's getting changed and where.

The way it is now, `onBuilderOptionsChanged` is only called in one place per view. I haven't had any issues with it, but if it becomes an issue we could easily swap back to having a function per field to update the query state.